core/state_transition.go

```diff
@@ -29,23 +29,24 @@ import (
 )
 
 /*
- * The State transitioning model
- *
- * A state transition is a change made when a transaction is applied to the current world state
- * The state transitioning model does all all the necessary work to work out a valid new state root.
- * 1) Nonce handling
- * 2) Pre pay / buy gas of the coinbase (miner)
- * 3) Create a new state object if the recipient is \0*32
- * 4) Value transfer
- * == If contract creation ==
- * 4a) Attempt to run transaction data
- * 4b) If valid, use result as code for the new state object
- * == end ==
- * 5) Run Script section
- * 6) Derive new state root
- */
+The State Transitioning Model
+
+A state transition is a change made when a transaction is applied to the current world state
+The state transitioning model does all all the necessary work to work out a valid new state root.
+
+1) Nonce handling
+2) Pre pay / buy gas from the sender
```

Comment:
The comment said "buy gas of the coinbase" before. Maybe you don't like "buy" in this context, but the comment was definitely incorrect before and is now slightly better because it actually reflects who is paying. The eth/gas exchange happens before running the transaction, in `StateTransition.BuyGas`.